### PR TITLE
feat: Create project with Che-server's SA on OpenShift with OAuth

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Create project with Che-server's SA on OpenShift with OAuth

### Screenshot/screencast of this PR
n/a

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/19717

### How to test this PR?
1. Deploy new che-server image `quay.io/skabashn/che-server:che19717_v5`
2. Enable `CHE_INFRA_OPENSHIFT_PROJECT_INIT__WITH__SERVER__SA=true`
3. Disable self-provisioning 
```
oc patch clusterrolebinding.rbac self-provisioners -p '{"subjects": null}'
oc patch clusterrolebinding.rbac self-provisioners -p '{ "metadata": { "annotations": { "rbac.authorization.kubernetes.io/autoupdate": "false" } } }'

```
4. Login as a regular user
5. Start a new workspace.
6. OpenshiftProject should be created sucessfully.
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties [hunk 1]
```diff
@@ -551,6 +551,10 @@ che.infra.openshift.route.labels=NULL
 # It has to be a valid DNS name.
 che.infra.openshift.route.host.domain_suffix=NULL
 
+# Initialize OpenShift project with Che server's service account
+# if OpenShift OAuth is enabled.
+che.infra.openshift.project.init_with_server_sa=false
```
Comment (sparkoo) [inline assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties:556]: why false? Shouldn't it be created with che-server sa by default?

**Review by sparkoo — APPROVED** [review]
lgtm. only question from me is what should be the default behavior

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties (hunk 1 above)
Comment (skabashnyuk) [reply assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties:556]: @l0rd wdyt what default value should be?

**Review by metlos — APPROVED** [review]

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties (hunk 1 above)
Comment (l0rd) [reply assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties:556]: `che-server` sa should be the default

**Review by mshaposhnik — APPROVED** [review]

**Comment by skabashnyuk** [comment]
[crw-ci-test --rebuild]

**Comment by skabashnyuk** [comment]
[crw-ci-test --rebuild]

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties [hunk 2]
```diff
@@ -550,6 +550,10 @@ che.infra.openshift.route.labels=NULL
 # It has to be a valid DNS name.
 che.infra.openshift.route.host.domain_suffix=NULL
 
+# Initialize OpenShift project with Che server's service account
```
Comment (themr0c) [inline assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties:553]:
```suggestion
# Initialize OpenShift project with {prod-short} server's service account
```

**Review by MichalMaler — APPROVED** [review]

**Pull request merged**